### DISCUSSION TITLE
IconTaskList

### DISCUSSION THREAD

**Discussion by ervinpopescu** [qtile/qtile#3065] [Ideas] [answered]
A widget that display just the icons of the current windows is a nice addition, imo. I am currently looking into the TaskList widget, but my expertise is really small, so I'm not getting too far with it :) It shouldn't be too hard to not display the text and just display the icon of the current window. If you could look into that, I'd really appreciate it.

**Comment by elParaguayo** [comment, marked as answer]
You could use the `TaskList` widget but change the text options to `""` and define a function which you can pass to `parse_text`:
https://github.com/qtile/qtile/blob/da93518b2ab924e803552decdb364d64de46088d/libqtile/widget/tasklist.py#L88-L95

Something like:
``` python
def no_text(text):
    return ""

...
widget.TaskList(
    parse_text=no_text,
    text_minimized="",
    text_maximized="",
    text_floating=""
)
```

**Reply by m-col** [reply]
Ha nice, very clever. Very hacky. I love it. I agree it might be a nice feature to have a shorthand version for that e.g. `icons_only=True`.

**Comment by ervinpopescu** [comment]
That's actually really nice. Too bad it would get obliterated from existence by the next release of qtile :) I'm gonna implement it anyways and hope you guys will implement that `icons_only` thing.

Edit: 
My bad, i read too fast :) It's not going to be deleted by the next release. Thanks a lot!!
